klab.engine/src/main/java/org/integratedmodelling/klab/documentation/extensions/table/AbstractTableArtifact.java

```diff
@@ -0,0 +1,98 @@
+package org.integratedmodelling.klab.documentation.extensions.table;
+
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.Map;
+
+import org.integratedmodelling.klab.api.documentation.views.IDocumentationView;
+import org.integratedmodelling.klab.api.documentation.views.ITableView;
+import org.integratedmodelling.klab.api.observations.IKnowledgeView;
+import org.integratedmodelling.klab.exceptions.KlabIOException;
+import org.integratedmodelling.klab.exceptions.KlabValidationException;
+import org.integratedmodelling.klab.provenance.Artifact;
+
+public abstract class AbstractTableArtifact extends Artifact implements IKnowledgeView {
+
+	private Map<String, ITableView> compiledViews = new HashMap<>();
+	
+	public static final String EXCEL_MEDIA_TYPE = "application/vnd.openxmlformats-officedocument.spreadsheetml.sheet";
+	public static final String HTML_MEDIA_TYPE = "text/html";
+    public static final String TEXT_PLAIN_MEDIA_TYPE = "text/markdown";
```

Comment:
I think this MIME types are swapped... TEXT = `text/plain` and MARKDOWN = `text/markdown`